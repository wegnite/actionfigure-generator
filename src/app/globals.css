@import "tailwindcss";
@import "./theme.css";

@plugin "tailwindcss-animate";

@layer base {
  * {
    @apply border-border outline-ring/50;
  }
  body {
    @apply bg-background text-foreground;
  }
  
  /* Smooth scrolling */
  html {
    scroll-behavior: smooth;
  }
  
  /* Selection colors */
  ::selection {
    @apply bg-primary/20 text-primary;
  }
}

/* Custom animations */
@keyframes gradient {
  0%, 100% {
    background-position: 0% 50%;
  }
  50% {
    background-position: 100% 50%;
  }
}

.animate-gradient {
  background-size: 200% 200%;
  animation: gradient 6s ease infinite;
}

/* actionFigure gradients */
.cf-gradient-primary {
  background: linear-gradient(135deg, 
    oklch(0.67 0.18 35.73) 0%, 
    oklch(0.75 0.25 78.45) 100%);
}

.cf-gradient-secondary {
  background: linear-gradient(135deg, 
    oklch(0.72 0.24 58.35) 0%, 
    oklch(0.80 0.22 78.45) 100%);
}

.cf-gradient-accent {
  background: linear-gradient(135deg, 
    oklch(0.82 0.15 58.35) 0%, 
    oklch(0.90 0.12 78.45) 100%);
}

.cf-gradient-light {
  background: linear-gradient(135deg, 
    oklch(0.94 0.05 58.35) 0%, 
    oklch(0.98 0.02 78.45) 100%);
}

.container {
  @apply mx-auto max-w-7xl px-4 md:px-8;
}

input,
select,
textarea {
  @apply border-border outline-ring/50 bg-background;
}

button {
  @apply cursor-pointer border-border outline-ring/50;
}

pre {
  @apply px-4;
}
