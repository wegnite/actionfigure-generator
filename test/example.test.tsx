// 示例测试文件 - 验证测试环境配置
import { render, screen } from '@testing-library/react'
import { describe, it, expect } from '@jest/globals'

// 简单组件用于测试
function TestComponent({ message }: { message: string }) {
  return (
    <div data-testid="test-component">
      <h1>测试组件</h1>
      <p>{message}</p>
    </div>
  )
}

describe('测试环境验证', () => {
  it('应该正确渲染React组件', () => {
    const message = '测试消息'
    render(<TestComponent message={message} />)
    
    // 验证组件正确渲染
    expect(screen.getByTestId('test-component')).toBeInTheDocument()
    expect(screen.getByText('测试组件')).toBeInTheDocument()
    expect(screen.getByText(message)).toBeInTheDocument()
  })

  it('应该支持DOM查询和匹配器', () => {
    render(<TestComponent message="Jest DOM 匹配器测试" />)
    
    // 验证 jest-dom 匹配器工作正常
    const heading = screen.getByText('测试组件')
    expect(heading).toBeVisible()
    expect(heading.tagName).toBe('H1')
  })

  it('应该支持快照测试', () => {
    const { container } = render(<TestComponent message="快照测试" />)
    
    // 快照测试确保组件输出一致性
    expect(container.firstChild).toMatchSnapshot()
  })
})

describe('测试工具集成验证', () => {
  it('应该支持异步测试', async () => {
    // 模拟异步操作
    const asyncFunction = async () => {
      return new Promise(resolve => setTimeout(() => resolve('异步完成'), 10))
    }

    const result = await asyncFunction()
    expect(result).toBe('异步完成')
  })

  it('应该支持模拟函数', () => {
    const mockFunction = jest.fn()
    mockFunction('测试参数')
    
    expect(mockFunction).toHaveBeenCalledWith('测试参数')
    expect(mockFunction).toHaveBeenCalledTimes(1)
  })
})