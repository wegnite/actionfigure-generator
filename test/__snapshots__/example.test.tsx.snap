// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`测试环境验证 应该支持快照测试 1`] = `
<div
  data-testid="test-component"
>
  <h1>
    测试组件
  </h1>
  <p>
    快照测试
  </p>
</div>
`;
